### PR TITLE
Multiple option proposal

### PR DESCRIPTION
**Pull request by andrewsource147**
Allow 2 types of proposal:
- Yes/no proposal, can be only queued if total vote meets quorum and vote yes is more than vote no
- Option proposal, can be only queued if total vote meets quorum

### CODE REVIEW

File: programs/govern/src/proposal.rs [hunk 1]
```diff
@@ -34,98 +31,40 @@ pub enum ProposalState {
...
-            VoteSide::Against => {
-                self.against_votes = unwrap_int!(self.against_votes.checked_add(vote_weight));
-            }
-            VoteSide::For => {
-                self.for_votes = unwrap_int!(self.for_votes.checked_add(vote_weight));
-            }
-            VoteSide::Abstain => {
-                self.abstain_votes = unwrap_int!(self.abstain_votes.checked_add(vote_weight));
-            }
+        if side > self.max_option {
```
Comment (codewithgun) [inline programs/govern/src/proposal.rs:59]: Should be `>=` ?
1. An OptionProposal created with 2 max options, which represent ABSENT, YES, NO. OptionProposal stored max_option = 2
2. Vote for NO, the index of NO is 2, it failed with InvalidVoteSide.

Comment (andrewsource147) [reply programs/govern/src/proposal.rs:59]: Abstain is 0, No is 2, then if they input side = 2, it will pass?

Comment (codewithgun) [reply programs/govern/src/proposal.rs:59]: Sorry my bad.

File: programs/govern/src/instructions/new_vote.rs [hunk 2]
```diff
@@ -36,8 +36,8 @@ impl<'info> NewVote<'info> {
         vote.voter = voter;
         vote.bump = bump;
 
-        vote.side = VoteSide::Pending.into();
-        vote.weight = 0;
+        vote.side = 0; // abstain vote
```
Comment (codewithgun) [inline programs/govern/src/instructions/new_vote.rs:39]: Can use ABSTAIN_VOTE_INDEX constant.

**Pull request merged**